src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java

```diff
@@ -87,73 +78,75 @@ public BasicAccessioningService(AccessionGenerator<MODEL, ACCESSION> accessionGe
      * @return
      */
-    private Map<ACCESSION, MODEL> saveAccessions(List<ModelHashAccession<MODEL, HASH, ACCESSION>> accessions) {
-        SaveResponse<ACCESSION, MODEL> response = basicAccessioningServiceSaveDelegate.doSaveAccessions(accessions);
+    private List<AccessionModel<MODEL, HASH, ACCESSION>> saveAccessions(List<AccessionModel<MODEL, HASH, ACCESSION>> accessions) {
+        SaveResponse<ACCESSION> response = basicAccessioningServiceSaveDelegate.doSaveAccessionedModels(accessions);
         accessionGenerator.postSave(response);
-        Map<ACCESSION, MODEL> savedAccessions = response.getSavedAccessions();
-        if (!response.getUnsavedAccessions().isEmpty()) {
-            savedAccessions.putAll(getUnsavedAccessions(response));
+
+        final List<AccessionModel<MODEL, HASH, ACCESSION>> savedAccessions = new ArrayList<>();
+        final List<AccessionModel<MODEL, HASH, ACCESSION>> saveFailedAccessions = new ArrayList<>();
+        accessions.stream().forEach(accessionModel -> {
+            if (response.isSavedAccession(accessionModel)) {
+                savedAccessions.add(accessionModel);
+            } else {
+                saveFailedAccessions.add(accessionModel);
+            }
+        });
+
+        if (!saveFailedAccessions.isEmpty()) {
+            List<AccessionModel<MODEL, HASH, ACCESSION>> unsavedAccessions = getUnsavedAccessions(saveFailedAccessions);
+            savedAccessions.addAll(unsavedAccessions);
+            dbService.enableAccessions(unsavedAccessions);
         }
+
         return savedAccessions;
     }
 
     /**
      * We try to recover all elements that could not be saved to return their accession to the user. This is only
      * expected when another application instance or thread has saved that element already with a different id. If
-     * all the elements could not be retrieved from the database it is considered an unexpected event and we throw
-     * {@link MissingUnsavedAccessions} to alert the system.
+     * any element can't be retrieved from the database we throw a {@link MissingUnsavedAccessions} to alert the system.
      *
-     * @param response
+     * @param saveFailedAccessions
      * @return
      */
-    private Map<ACCESSION, MODEL> getUnsavedAccessions(SaveResponse<ACCESSION, MODEL> response) {
-        List<MODEL> unsavedObjects = new ArrayList<>(response.getUnsavedAccessions().values());
-        final Map<ACCESSION, MODEL> unsavedObjectAccessions = getAccessions(unsavedObjects);
-        if (unsavedObjectAccessions.size() != unsavedObjects.size()) {
-            throw new MissingUnsavedAccessions(unsavedObjectAccessions, unsavedObjects);
+    private List<AccessionModel<MODEL, HASH, ACCESSION>> getUnsavedAccessions(
+            List<AccessionModel<MODEL, HASH, ACCESSION>> saveFailedAccessions) {
+
+        Set<HASH> saveFailedHashes = saveFailedAccessions.stream().map(AccessionModel::getHash)
+                .collect(Collectors.toSet());
```

Comment:
If the other renaming suggestion in this class is followed, then `unsavedHashes` would make all the names homogeneous.